vortex-array/src/arrays/chunked/compute/elementwise.rs

```diff
@@ -11,25 +11,40 @@ impl ChunkedArray {
         compute_fn: &ComputeFn,
         args: &InvocationArgs,
     ) -> VortexResult<Option<Output>> {
-        assert!(compute_fn.is_elementwise());
+        assert!(
+            compute_fn.is_elementwise(),
+            "Expected elementwise compute function"
+        );
+        assert!(
+            !args.inputs.is_empty(),
+            "Elementwise compute function requires at least one input"
+        );
 
-        // We know the first input argument is ourselves, if there is one other input argument
-        // then we can delegate the compute function. Otherwise, we return None.
-        if args.inputs.len() != 2 || args.inputs[1].array().is_none() {
+        // If not all inputs are arrays, then we pass.
+        if args.inputs.iter().any(|a| a.array().is_none()) {
             return Ok(None);
         }
-        let rhs = args.inputs[1].array().vortex_expect("checked already");
 
         let mut idx = 0;
         let mut chunks = Vec::with_capacity(self.nchunks());
+        let mut inputs = Vec::with_capacity(args.inputs.len());
 
         for chunk in self.non_empty_chunks() {
-            let sliced = slice(rhs, idx, idx + chunk.len())?;
+            inputs.clear();
+            inputs.push(chunk.clone());
+            for i in 1..args.inputs.len() {
+                let input = args.inputs[i].array().vortex_expect("checked already");
+                let sliced = slice(input, idx, idx + chunk.len())?;
+                inputs.push(sliced);
+            }
```

Comment:
How about we remove the clear, push chunk first outside of the loop and here do `inputs[i] = ...`